69 ribo seq integration

alexgrent opened the pull request:
PR is about the integration of RiboSeq data within the ReactomeGSA backend
To handle RiboSeq data 2 files are needed one Ribo seq file and one RNA-seq file. It is assumed that both files habe matching columns. Further one metadata file provided by the client. 

This includes the following implementation steps:
- Implementing a new API endpoint to handle 2 files further modifiying existing analysis endpoint for specific metadata handling
- Integrate new Pathway algorithm, which is a modified version of PADOG

alexgrent committed:
Add internal structure for Ribo Seq data analysis

alexgrent committed:
rm Method and ref External data

alexgrent committed:
add new riboseq_counts

alexgrent committed:
changes PADOG methos description

alexgrent committed:
integrated ribo seq support

alexgrent committed:
add example

alexgrent committed:
add condition to riboseq example

alexgrent committed:
ch. def. para

alexgrent committed:
ch. format of data

alexgrent committed:
rm id

Gionmattia committed:
reformat file

alexgrent committed:
add BE approach for Ribo seq

Gionmattia committed:
Improv riboseq_analyser.R

Gionmattia committed:
Merge remote-tracking branch 'origin/69-riboSeq-integration' into 69-riboSeq-integration

alexgrent committed:
BE adaption ribo seq

alexgrent committed:
Merge remote-tracking branch 'origin/69-riboSeq-integration' into 69-riboSeq-integration

alexgrent committed:
ch. identifier

Gionmattia committed:
Comment ref

Gionmattia committed:
Merge remote-tracking branch 'origin/69-riboSeq-integration' into 69-riboSeq-integration

Gionmattia committed:
Code Ref

alexgrent committed:
install libs depend

jgriss commented on reactome_analysis_worker/reactome_analysis_worker/resources/r_code/riboseq_analyser.R:
```diff
@@ -0,0 +1,671 @@
+# Implementation of TerraPadog algorithm
+
+
+terapadog <- function (esetm = NULL, exp_de = NULL, paired = FALSE,
```
Having the complete analysis code here is not ideal. This increases the complexity of our worker code quite a bit.

`terapadog` should be loaded from its own package. In case there is no package yet, there needs to be a discussion how this code will be maintained.

alexgrent replied:
I totally agree, with you. In case of packaging, maintaining could be done externally. (But will not be in our control) 
But generally, a Bioconductor package will be done anyways by @Gionmattia i guess.

jgriss reviewed (changes requested):
Hi @alexgrent 

Thanks a lot for this contribution!

My main concern is that the terapadog logic was copied into our analyser. This adds a new level of complexity to our code.

Ideally, the terapadog logic should be put in its own package which we can then load into our analyser

alexgrent commented on reactome_analysis_api/reactome_analysis_api/__main__.py:
```diff
@@ -80,6 +83,133 @@ def custom_abort(code: int, message: str):
+
+@app.route("/uploadRibo", methods=["POST"])
+def upload_ribo():
+    store_file = request.args.get('store', 'true').lower() == "true"
```
this is similar used to the the /upload endpoint as in line 211. As far as I understood is that it is possible to set this parameter depending on the client.

alexgrent commented on reactome_analysis_api/reactome_analysis_api/__main__.py:
```diff
@@ -80,6 +83,133 @@ def custom_abort(code: int, message: str):
+        (name, ribo_extension) = os.path.splitext(user_file_ribo)
+
+        if rna_extension == ".xlsx" or ribo_extension == ".xlsx":
+            UPLOAD_ERRORS.labels(extension=rna_extension + ribo_extension).inc()
```
changed to xslx but I would agree with you since the user base is not that big and we do not perform any analytics on basis of the data the counter is not necessar.

alexgrent committed:
ch. acc. rev

alexgrent committed:
rm redundant code

alexgrent commented on reactome_analysis_api/reactome_analysis_api/__main__.py:
```diff
@@ -80,6 +83,133 @@ def custom_abort(code: int, message: str):
+        LOGGER.info("Empty file uploaded.")
+        return custom_abort(400, "The uploaded file seems to be empty.")
+
+    # guess the delimiter
```
Done

alexgrent committed:
rm terapadog logic and use bioconductor package for terapadog instead

alexgrent committed:
outsources assign_Regmode to bioconductor package

alexgrent committed:
ch. pack. name

alexgrent committed:
ch.

alexgrent committed:
Update analysis_controller.py

jgriss committed:
Merge branch 'develop' into 69-riboSeq-integration

jgriss committed:
Merge remote-tracking branch 'refs/remotes/origin/69-riboSeq-integration' into 69-riboSeq-integration

jgriss committed:
[api] Adapted for new connexion version

jgriss committed:
[api] Updated API added data_types field to method

This lists the supported data types for each analysis method. The reason for this change is to support methods that may only be compatible with some data types

jgriss committed:
[api] Adapted API for new terapadog method

jgriss committed:
Merge branch 'develop' into 69-riboSeq-integration

jgriss committed:
[api] Adapted Ribo-seq example description

jgriss committed:
[api/datasets] RiboSeq requires metadata for all samples

jgriss committed:
[api] Adapted ExternalData specification for RiboSeq

jgriss committed:
[datasets] Fixed RiboSeq example data type

jgriss committed:
[worker/datasets/testsuite] Completed terapdog support

jgriss committed:
Merge branch 'develop' into 69-riboSeq-integration

jgriss committed:
[api] Adapted changes for the terapadog update

- Added "data_types" to ExternalDataSource
- Added "doc_link" to DataType
- Added example implementation of doc_link for the ribo_rna_seq DataType
-  Fixed data type for the RiboSeq example
- Added data_types to all ExternalDataSources
